frontend/ui/widget/inputtext.lua

```diff
@@ -595,11 +595,11 @@ function InputText:onKeyPress(key)
     end
     local handled = true
 
-    if not key["Ctrl"] and not key["Shift"] and not key["Alt"] then
+    if not key["Ctrl"] and not key["Shift"] and not key["Alt"] and not key["ScreenKB"] then
         if key["Backspace"] then
             self:delChar()
         elseif key["Del"] then
-            self:delNextChar()
+            self:delChar()
```

Comment:
This affects the emulator, when using a real keyboard. On PCs, backspace remove the previous char, Del/Suppr remove the next char. No reason to change that behaviour.

Reply:
Kindle has a “del” key only, no backspace. Which means you can’t delete with the delete key.

Reply:
would this be better ...

```lua
if Device:hasSymKey() then
        if key["Del"] then
            self:delChar()
        end
    else
        if key["Del"] then
            self:delNextChar()
        end
    end
```

Reply:
Not `Device:hasScreenKB() or Device:hasSymKey()`?

Reply:
no `delete` key on k4 so the first bit (`hasScreenKB`) is unnecessary, that is why I am using `ScreenKB + Back` as delete there

Reply:
Should be good then